aep/general/0214/aep.md.j2

```diff
@@ -1,5 +1,79 @@
 # Resource expiration
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/19) for more
-information.
+Customers often want to provide the time that a given resource or resource
+attribute is no longer useful or valid (e.g. a rotating security key). Currently
+we recommend that customers do this by specifying an exact "expiration time"
+into a `google.protobuf.Timestamp expire_time` field; however, this adds
+additional strain on the user when they want to specify a relative time offset
+until expiration rather than a specific time until expiration.
+
+Furthermore, the world understands the concept of a "time-to-live", often
+abbreviated to TTL, but the typical format of this field (an integer, measured
+in seconds) results in a sub-par experience when using an auto-generated client
+library.
+
+## Guidance
+
+1.  APIs wishing to convey an expiration **must** rely on a
+    [`google.protobuf.Timestamp`][timestamp] field called `expire_time`.
+2.  APIs wishing to allow a relative expiration time must define a `oneof`
+    called `expiration` (or `{something}_expiration`) containing both the
+    `expire_time` field and a separate [`google.protobuf.Duration`][duration]
+    field called `ttl`, the latter marked as input only.
+3.  APIs **must** always return the expiration time in the `expire_time` field
+    and leave the `ttl` field blank when retrieving the resource.
+4.  APIs that rely on the specific semantics of a "time to live" (e.g., DNS
+    which must represent the TTL as an integer) **may** use an `int64 ttl`
+    field (and **should** provide an [aep.dev/not-precedent](./0200.md) comment
+    in this case).
+
+### Example
+
+{% tab oas %}
+
+```proto
+message ExpiringResource {
+  // google.api.resource and other annotations and fields
+
+  oneof expiration {
+    // Timestamp in UTC of when this resource is considered expired.
+    // This is *always* provided on output, regardless of what was sent
+    // on input.
+    google.protobuf.Timestamp expire_time = 2;
+
+    // Input only. The TTL for this resource.
+    google.protobuf.Duration ttl = 3 [(google.api.field_behavior) = INPUT_ONLY];
+  }
+}
+```
```

Comment:
```suggestion
{% tab proto %}

```proto
message ExpiringResource {
  // google.api.resource and other annotations and fields

  oneof expiration {
    // Timestamp in UTC of when this resource is considered expired.
    // This is *always* provided on output, regardless of what was sent
    // on input.
    google.protobuf.Timestamp expire_time = 2;

    // Input only. The TTL for this resource.
    google.protobuf.Duration ttl = 3 [(google.api.field_behavior) = INPUT_ONLY];
  }
}
\```
{% tab oas %}

```